common/djangoapps/third_party_auth/models.py

```diff
@@ -241,22 +243,31 @@ def get_register_form_data(cls, pipeline_kwargs):
             values for that field. Where there is no value, the empty string
             must be used.
         """
+        current_provider = kwargs.pop('current_provider', None)
+        registration_form_data = {}
+
         # Details about the user sent back from the provider.
         details = pipeline_kwargs.get('details')
+        registration_form_data['email'] = details.get('email', '')
+        registration_form_data['name'] = details.get('fullname', '')
+
+        # There are often nonstandard requirements for some registration flows,
+        # like the user's country code.
+        if current_provider.required_extra_attributes:
```

Comment:
If we leave `current_provider` as an optional kwarg, here's where we get our first AttributeError when we don't pass it.